test_framework/scripts/longhorn-setup.sh

```diff
@@ -25,16 +25,22 @@ set_kubeconfig_envvar(){
   ARCH=${1}
   BASEDIR=${2}
 
-    if [[ ${ARCH} == "amd64" ]] ; then
+  if [[ ${ARCH} == "amd64" ]] ; then
     if [[ ${TF_VAR_k8s_distro_name} == [rR][kK][eE] ]]; then
       export KUBECONFIG="${BASEDIR}/kube_config_rke.yml"
     elif [[ ${TF_VAR_k8s_distro_name} == [rR][kK][eE]2 ]]; then
-      export KUBECONFIG="${BASEDIR}/terraform/aws/${DISTRO}/rke2.yaml"
+      export KUBECONFIG="${BASEDIR}/terraform/${LONGHORN_TEST_CLOUDPROVIDER}/${DISTRO}/rke2.yaml"
+    elif [[ ${TF_VAR_k8s_distro_name} == "gke" ]]; then
+      gcloud container clusters get-credentials `terraform -chdir=${TF_VAR_tf_workspace}/terraform/${LONGHORN_TEST_CLOUDPROVIDER}/${TF_VAR_k8s_distro_name} output -raw cluster_name` --zone `terraform -chdir=${TF_VAR_tf_workspace}/terraform/${LONGHORN_TEST_CLOUDPROVIDER}/${TF_VAR_k8s_distro_name} output -raw cluster_zone` --project ${TF_VAR_gcp_project}
     else
-      export KUBECONFIG="${BASEDIR}/terraform/aws/${DISTRO}/k3s.yaml"
+      export KUBECONFIG="${BASEDIR}/terraform/${LONGHORN_TEST_CLOUDPROVIDER}/${DISTRO}/k3s.yaml"
     fi
   elif [[ ${ARCH} == "arm64"  ]]; then
-    export KUBECONFIG="${BASEDIR}/terraform/aws/${DISTRO}/k3s.yaml"
+    if [[ ${TF_VAR_k8s_distro_name} == "gke" ]]; then
+      gcloud container clusters get-credentials `terraform -chdir=${TF_VAR_tf_workspace}/terraform/${LONGHORN_TEST_CLOUDPROVIDER}/${TF_VAR_k8s_distro_name} output -raw cluster_name` --zone `terraform -chdir=${TF_VAR_tf_workspace}/terraform/${LONGHORN_TEST_CLOUDPROVIDER}/${TF_VAR_k8s_distro_name} output -raw cluster_zone` --project ${TF_VAR_gcp_project}
```

Comment:
note: `It takes a project and a zone as parameters, passed through by set defaults or flags. By default, credentials are written to HOME/.kube/config`

ref: https://cloud.google.com/sdk/gcloud/reference/container/clusters/get-credentials